index.js

```diff
@@ -49,17 +49,22 @@ function createStore (modules) {
     if (events[event]) {
       var changes = { }
       var changed, key
+      var newState = { }
+
+      for (key in state) newState[key] = state[key]
+
       events[event].forEach(function (i) {
         var diff = i(state, data)
-        if (diff) {
+        for (key in diff) {
           changed = true
-          var newState = { }
-          for (key in state) newState[key] = state[key]
-          for (key in diff) newState[key] = changes[key] = diff[key]
-          state = newState
```

Comment:
We are changing behavior here. If 2+ listeners will listen for one event, the second listener will receive old state and could override first listener changes.

Reply:
```
        var diff = i(newState, data)
```

Changing #L57 to this could fix getting old state.

Reply:
Cool idea. But `store.get()` will give old result. It is OK if there is a reason for it (are we reducing the size).

Reply:
Inside a reducer, the first argument will be the intermediate uncommitted state.
`store.get()` will give you the committed previous state.

Can it be a feature?
Just throwing out random ideas.

Reply:
> Can it be a feature?

Dangerous :D

> Just throwing out random ideas.

Were these changes reduce JS size?

Reply:
> Were these changes reduce JS size?

The size of `index.js` reduced to `174 B`, and `index.js` & `hook` together to `332 B`.

Reply:
1 byte is not enough for these changes.

Can we undo it? Do you need it for Promise support?

Reply:
> Cool idea. But `store.get()` will give old result. It is OK if there is a reason for it (are we reducing the size).

As we are already passing current state to the reducer, is `store.get()` supposed to be called inside it?
Ideally, reducers are pure functions.

Reply:
> is store.get() supposed to be called inside it?

It could happen in async event listeners (of course, not in simple reducers). I agree that reducers should be pure.

(BTW, I am still thinking how to split “event listeners” and reducers)